src/core/tasks.py

```diff
@@ -0,0 +1,140 @@
+from __future__ import absolute_import
+import csv
+import ftplib
+import os
+
+from celery import task
+from django.conf import settings
+
+from core import models, email
+from services import ServiceHandler, JuraUpdateService
+
+
+def _read_csv(path):
+    """Read a CSV and return its rows, cleaning NUL bytes."""
+    old_csv = open(path, 'rb')
+    data = old_csv.read()
+    old_csv.close()
+    if path != 'bisac.csv':
+        os.remove(path)
+    new_csv = open('new.csv', 'wb')
+    new_csv.write(
+        data.replace('\00', '')
+    )
+    new_csv.close()
+
+    with open('new.csv', 'rb') as f:
+        reader = csv.reader(
+            f.read().splitlines()
+        )
+        return reader
+
+
+@task(name='add-metadata')
+def add_metadata():
+    """Adds book metadata from CSVs on an FTP site.
+
+    Accesses a specified FTP dir, copies CSVs to
+    the local dir then deletes them from FTP, reads
+    the CSV and matches ISBN13s to Rua books and
+    updates their metadata with the matching CSV
+    column values, and sends a notification email.
+    """
+    if settings.FTP_METADATA_FEATURE:
+        service = ServiceHandler(service=JuraUpdateService)
+
+        ftp = ftplib.FTP(getattr(settings, 'FTP_URL', None))
```

Comment:
```with ftplib.FTP(getattr(settings, 'FTP_URL', None)) as ftp:```

It can be used with the ```with statement``` after all. Try using in a python shell ```help(ftplib.FTP)```

Reply:
I tried this but it threw a tantrum so I've reverted it

Reply:
Interesting, well it doesn't matter that much